ow_power_system/src/PowerSystemNode.cpp

```diff
@@ -78,65 +80,97 @@ bool PowerSystemNode::loadSystemConfig()
   return true;
 }
 
-PrognoserVector PowerSystemNode::loadPowerProfile(const string& filename)
+PrognoserVector PowerSystemNode::loadPowerProfile(const string& filename, string csv)
 {
   ifstream file(filename);
+
+  // Checking for .csv extension isn't strictly necessary as long as the
+  // formatting is correct, but for now, CSV files are expected.
+  if (filename.substr(filename.size() - 4) != ".csv")
+  {
+    throw ERR_INVALID_FILE_NAME;
+  }
```

Comment:
This can be removed -- I see no need to enforce a filename.

Reply:
Glad we agree. Just want to chime in that if the ".csv" check is removed then the print statements on lines 356, 360, 410 should be made agnostic to the file extension length.